rust-client/src/main.rs

```diff
@@ -47,14 +47,21 @@ struct Opt {
     #[clap(long)]
     secret_key_seed: u8,
 
+    /// If set to `true`, use relay-v1 protocol.
+    /// Per default relay-v2 is used.
     #[clap(long)]
     relay_v1: bool,
+
+    /// Api-key used to authenticate our client at the server.
+    #[clap(long)]
+    api_key: String,
```

Comment:
I don't think we should pass a secret via a command line argument. This is problematic due to:

- The secret ends up in the shell's history.
- Other users can see the secret via `/proc`.

Instead I would use either an environment variable or a file where the file path can be passed as an argument.

What do you think @elenaf9?

Reply:
Good point. Reading it from env variable now (50253b0). If it is not present I am logging a warning but still continuing, in case that we ever decide to (temporarily?) disable auth again. Can also change it to throw an error, if you think that makes more sense.

Reply:
Given that it is a mechanism for the server to establish trust to the client and not the other way around, I think defaulting to not using the API key is fine here. :+1: